src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -364,7 +365,19 @@ private void walk(TypedVar symbol, boolean isDefault) {
           }
         }
 
-        visitObjectType(ftype, ftype.getPrototype());
+        if (ftype.isInterface()) {
+          visitObjectType(ftype, ftype.getPrototype(), false);
+
+          if (Iterables.any(ftype.getOwnPropertyNames(), NOT_PROTOTYPE)) {
```

Comment:
you can use Predicates.not(Predicates.equalTo("prototype"))
and static import to get
not(equalTo("prototype"))
opt: static import for "any" as well